qa/common/util.sh

```diff
@@ -468,6 +468,16 @@ function kill_servers () {
     done
 }
 
+# Upload a local directory to a GCS path
+function gcs_upload () {
+    local local_path=$1
+    local gcs_path=$2
+    rm -f gcs_upload.gsutil_cp.log
+    until gsutil -m cp -c -L gcs_upload.gsutil_cp.log -r $local_path $gcs_path; do
```

Comment:
I'm afraid that `-m` flag by default includes `-c` flag. 
The issue with it that it will keep return non-zero code if error happen.
>-c
>    If an error occurs, continue attempting to copy the remaining files. If any copies are unsuccessful, gsutil's exit status is non-zero, even if this flag is set. This option is implicitly set when running gsutil -m cp....
> https://cloud.google.com/storage/docs/gsutil/commands/cp

Reply:
may be we should drop this flag in favor of below?
`gsutil cp -L gcs_upload.gsutil_cp.log -r $local_path $gcs_path`

Reply:
I think the `-c` flag is exactly what we want, because the goal is to capture the failure until it success. (or until a max retry which I am looking into that)